test_config_files.py

```
@@ -0,0 +1,523 @@
+        os.chmod(config_file, 0o000)
+        
+        try:
+            with pytest.raises(PermissionError):
+                with open(config_file, 'r') as f:
+                    json.load(f)
+        finally:
+            # Restore permissions for cleanup
+            os.chmod(config_file, 0o644)
+    
+    @patch('builtins.open', side_effect=IOError("Simulated IO error"))
+    def test_io_error_handling(self, mock_open):
```

Flake8 (7.2.0)

[error] redefinition of unused 'mock_open' from line 7

(F811)